### DISCUSSION TITLE
Pipeline input parameters

### DISCUSSION THREAD

**Discussion by annhov** [elyra-ai/elyra#2985] [Pipeline Editor/ Kubeflow Pipelines + Apache Airflow]
Hi everyone! 

I have a use case where I need to create a kubeflow pipeline from elyra, export it to kubeflow server and then run it with different parameters using[ kfp api ](https://www.kubeflow.org/docs/components/pipelines/v1/reference/api/kubeflow-pipeline-api-spec/). 
This might be related to the previous discussions #2970 and #2921, but as the development is still in the planning stage/postponed and the branch from the discussion (https://github.com/lresende/elyra/commits/dev-parameter) is outdated, I'd like to modify the source code so that it will be possible to receive input parameters. The handling of these params can be done by just setting some new environment variables (maybe from bootstrapper.py) and doesn't need to include UI, the only required part is the modification of pipeline/yaml generation.

Maybe there is already some development branch I can refer to? Or can you please suggest which parts should be changed to implement this feature?

**Comment by ptitzler** [comment]
Hi @annhov! We are currently implementing changes to the Kubeflow Pipelines code generator that will be [included in 3.13](https://github.com/elyra-ai/elyra/issues/2986). This rework is required to properly support pipeline parameters, will should be ready in time for the 3.14 release.

**Reply by annhov** [reply]
Thanks for the response @ptitzler. Are there any estimates for the 3.13/3.14 versions?

**Reply by ptitzler** [reply]
We haven't identified [milestone dates yet](https://github.com/elyra-ai/elyra/milestones), but in the past we've tried to publish a minor release once a month. If we can stick to that for the two releases, 3.14 should be available sometime in December 2022.

**Reply by ptitzler** [reply]
FYI: pipeline parameter support was delivered in https://github.com/elyra-ai/elyra/releases/tag/v3.14.0. Several fixes went into subsequent patch releases.
